### PR TITLE
SDR service API

### PR DESCRIPTION
**Pull request by agafox**

### CODE REVIEW

File: restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreterParams.java [hunk 1]
```diff
@@ -59,7 +60,7 @@
     private String transferor;
     private String transferee;
 
-    private VoiceInterpreterParams(Configuration configuration, DaoManager storage, ActorRef callManager, ActorRef conferences, ActorRef bridgeManager, ActorRef smsService, Sid account, Sid phone, String version, URI url, String method, URI fallbackUrl, String fallbackMethod, URI statusCallback, String statusCallbackMethod, String referTarget, String emailAddress, ActorRef monitoring, String rcml, boolean asImsUa, String imsUaLogin, String imsUaPassword, String transferor, String transferee) {
+    private VoiceInterpreterParams(Configuration configuration, DaoManager storage, ActorRef callManager, ActorRef conferences, ActorRef bridgeManager, ActorRef smsService, Sid account, Sid phone, String version, URI url, String method, URI fallbackUrl, String fallbackMethod, URI statusCallback, String statusCallbackMethod, String referTarget, String emailAddress, ActorRef monitoring, ActorRef sdr, String rcml, boolean asImsUa, String imsUaLogin, String imsUaPassword, String transferor, String transferee) {
```
Comment (jaimecasero) [inline restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreterParams.java]: i know this is not particular to this PR,but this constructor argument size is getting too long. We may consider to create a simple class to wrap all of this...

File: restcomm/restcomm.sdr.api/src/main/java/org/restcomm/connect/sdr/api/B2BUASdrMessage.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+package org.restcomm.connect.sdr.api;
+
+import org.restcomm.connect.commons.dao.Sid;
+
+import javax.servlet.sip.SipServletMessage;
+
+/**
+ * @author oleg.agafonov@telestax.com (Oleg Agafonov)
+ */
+public class B2BUASdrMessage {
```
Comment (jaimecasero) [inline restcomm/restcomm.sdr.api/src/main/java/org/restcomm/connect/sdr/api/B2BUASdrMessage.java]: consider final class?

File: restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java [hunk 3]
```diff
@@ -476,7 +476,7 @@ private boolean isOutbound() {
             } else {
                 to = ((TelURL) this.to).getPhoneNumber();
             }
-            final CallInfo info = new CallInfo(id, external, type, direction, created, forwardedFrom, name, from, to, invite, lastResponse, webrtc, muted, isFromApi, callUpdatedTime, mediaAttributes);
+            final CallInfo info = new CallInfo(id, accountId, external, type, direction, created, forwardedFrom, name, from, to, invite, lastResponse, webrtc, muted, isFromApi, callUpdatedTime, mediaAttributes);
```
Comment (jaimecasero) [inline restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java]: same comment as before about number of arguments.Also make sure all tests in all modules are using new constructor

File: restcomm/restcomm.application/src/main/java/org/restcomm/connect/application/Bootstrapper.java [hunk 4]
```diff
@@ -246,48 +244,66 @@ public UntypedActor create() throws Exception {
...
+        final String className = configuration.getString("sdr-service[@class]");
+        if (className != null) {
+            final Props props = new Props(new UntypedActorFactory() {
+                @Override
+                public Actor create() throws Exception {
+                    return (Actor) new ObjectFactory(loader).getObjectInstance(className);
+                }
+            });
+            ActorRef sdr = system.actorOf(props);
+            sdr.tell(new StartSdrService(configuration), null);
```
Comment (jaimecasero) [inline restcomm/restcomm.application/src/main/java/org/restcomm/connect/application/Bootstrapper.java]: do we need some explicit cleanup of this service on application closing? do we need to send some message or close the actor on application closing?

Comment (agafox) [reply restcomm/restcomm.application/src/main/java/org/restcomm/connect/application/Bootstrapper.java]: Not here for sure. akka actors have their own close context mechanizm.

File: restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java (hunk 3 above)
Comment (agafox) [reply restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java]: Fixed in all tests as well

**Comment by jaimecasero** [comment]
@agafox  apart form some simple consdierations this PR looks ok in terms of design/code. Please go through comments provided.

On testsuite results,this is current one:
https://cxs.restcomm.com/view/VVSSquad/view/ConnectCommunity/job/Restcom-CI-Multibranch/view/change-requests/job/PR-2689/5/testReport/

i think they are just unstable tests we need to mark.This is also failing in master form time to time
https://cxs.restcomm.com/view/VVSSquad/view/ConnectCommunity/job/Restcom-CI-Multibranch/job/master/41/testReport/junit/org.restcomm.connect.testsuite.telephony/DialRecordingS3UploadSecureTest/CITestsuiteSeq___testRecordCall/

Could you please make sure this 3 tests are workign locally?

File: restcomm/restcomm.sdr.api/src/main/java/org/restcomm/connect/sdr/api/B2BUASdrMessage.java (hunk 2 above)
Comment (agafox) [reply restcomm/restcomm.sdr.api/src/main/java/org/restcomm/connect/sdr/api/B2BUASdrMessage.java]: fixed

**Comment by agafox** [comment]
All failed tests passed locally

**Pull request merged**